.github/tekton/pull-request-pipeline.yaml

```diff
@@ -0,0 +1,138 @@
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: ks-releaser-pull-request
+spec:
+  workspaces:
+    - name: repo
+  params:
+    - name: repo-full-name
+      description: "Repository full name. like: kubesphere-sigs/ks-releaser"
+    - name: clone-url
+      description: Repository URL to clone from.
+    - name: revision
+      description: Revision to checkout. (branch, tag, sha, ref, etc...)
+      default: master
+    - name: dashboard-url
+      description: Tekton dashboard access URL, like http://demo:31962/#/namespaces/ks/pipelineruns.
+  tasks:
+    - name: set-running-status
+      taskRef:
+        name: github-set-status
+      params:
+        - name: REPO_FULL_NAME
+          value: $(params.repo-full-name)
+        - name: SHA
+          value: $(params.revision)
+        - name: DESCRIPTION
+          value: Build has started
+        - name: STATE
+          value: pending
+        - name: TARGET_URL
+          value: $(params.dashboard-url)
+    - name: checkout
+      taskRef:
+        name: git-clone
+      params:
+        - name: url
+          value: $(params.clone-url)
+        - name: revision
+          value: $(params.revision)
+      workspaces:
+        - name: output
+          workspace: repo
+    - name: build
+      runAfter:
+        - checkout
+      taskRef:
+        name: goreleaser-and-trivy
```

Comment:
We already put the common tasks into ks-infra. It's necessary to let all contributors know that.

Reply:
I've added a document about those common task, please refer to https://github.com/kubesphere-sigs/ks-infra/tree/master/prod/ks-devops-ext-tekton-common#common-tasks.